protocol/x/subaccounts/types/subaccount.go

```diff
@@ -73,26 +73,25 @@ func (m *Subaccount) GetUsdcPosition() *big.Int {
 }
 
 // SetUsdcAssetPosition sets the balance of the USDC asset position to `newUsdcPosition`.
-// If the absolute value of `newUsdcPosition` cannot be represented in a uint64,
-// an error is returned.
-func (m *Subaccount) SetUsdcAssetPosition(newUsdcPosition *big.Int) error {
-	if m != nil {
-		usdcAssetPosition := m.getUsdcAssetPosition()
-		if newUsdcPosition == nil || newUsdcPosition.Sign() == 0 {
-			if usdcAssetPosition != nil {
-				m.AssetPositions = m.AssetPositions[1:]
-			}
-		} else {
-			if usdcAssetPosition == nil {
-				usdcAssetPosition = &AssetPosition{
-					AssetId: assettypes.AssetUsdc.Id,
-				}
-				m.AssetPositions = append([]*AssetPosition{usdcAssetPosition}, m.AssetPositions...)
+func (m *Subaccount) SetUsdcAssetPosition(newUsdcPosition *big.Int) {
+	if m == nil {
+		return
```

Comment:
Should we panic / error log if this ever happens?

It doesn't seem like this function should silently fail if called with a nil pointer.

Reply:
I think this was meant to be a convenience method for tests only. I don't think there is a reason to use this instead of `UpdateAssetPositions`. wdyt?

Reply:
It's currently used in subaccounts keeper [here](https://github.com/dydxprotocol/v4-chain/blob/dcd6ecb083145ccc4021aef8bbbcb2a3897f7111/protocol/x/subaccounts/keeper/subaccount.go#L436)

Created a ticket [here](https://linear.app/dydx/issue/CLOB-993/remove-setusdcassetposition-function) to remove it later.